skfda/exploratory/visualization/representation.py

```diff
@@ -119,87 +113,173 @@ def _get_color_info(
+        *,
+        fig: Optional[Figure] = None,
+        axes: Optional[Axes] = None,
+        n_rows: Optional[int] = None,
+        n_cols: Optional[int] = None,
+        n_points: Union[int, Tuple[int, int], None] = None,
+        domain_range: Optional[DomainRangeLike] = None,
+        group: Optional[Sequence[K]] = None,
+        group_colors: Optional[Indexable[K, ColorLike]] = None,
+        group_names: Optional[Indexable[K, str]] = None,
+        colormap: Union[Colormap, str, None] = 'autumn',
+        legend: bool = False,
+        **kwargs: Any,
     ) -> None:
+        BasePlot.__init__(self)
         self.fdata = fdata
-        self.gradient_values = gradient_values
-        if self.gradient_values is not None:
-            if len(self.gradient_values) != fdata.n_samples:
+        self.gradient_criteria = gradient_criteria
+        if self.gradient_criteria is not None:
+            if len(self.gradient_criteria) != fdata.n_samples:
                 raise ValueError(
-                    "The length of the gradient color"
-                    "list should be the same as the number"
+                    "The length of the gradient color",
+                    "list should be the same as the number",
                     "of samples in fdata",
                 )
 
             if min_grad is None:
-                self.min_grad = min(self.gradient_values)
+                self.min_grad = min(self.gradient_criteria)
             else:
                 self.min_grad = min_grad
 
             if max_grad is None:
-                self.max_grad = max(self.gradient_values)
+                self.max_grad = max(self.gradient_criteria)
             else:
                 self.max_grad = max_grad
 
             aux_list = [
                 grad_color - self.min_grad
-                for grad_color in self.gradient_values
+                for grad_color in self.gradient_criteria
             ]
 
-            self.gradient_list = (
+            self.gradient_list: Sequence[float] = (
                 [
                     aux / (self.max_grad - self.min_grad)
                     for aux in aux_list
                 ]
             )
         else:
-            self.gradient_list = []
+            self.gradient_list = None
+
+        self.n_points = n_points
+        self.domain_range = domain_range
+        self.group = group
+        self.group_colors = group_colors
+        self.group_names = group_names
+        self.legend = legend
+        self.colormap = colormap
+
+        if self.gradient_list is None:
+            sample_colors, patches = _get_color_info(
+                self.fdata,
+                self.group,
+                self.group_names,
+                self.group_colors,
+                self.legend,
+                kwargs,
+            )
+        else:
+            patches = None
+            colormap = matplotlib.cm.get_cmap(self.colormap)
+            colormap = colormap.reversed()
```

Comment:
Why reversed?

Reply:
i changed it, in the autumn case its better in this way, will just reverse it if no colormap is specified